botocore/utils.py

```diff
@@ -1085,3 +1085,54 @@ def _get_response(self, full_url, headers, timeout):
 
     def full_url(self, relative_uri):
         return 'http://%s%s' % (self.IP_ADDRESS, relative_uri)
+
+
+def get_configured_signature_version(service_name, client_config,
+                                     scoped_config):
+    """
+    Gets the manually configured signature version.
+
+    :returns: the customer configured signature version, or None if no
+        signature version was configured.
+    """
+    # Client config overrides everything.
+    if client_config and client_config.signature_version is not None:
+        return client_config.signature_version
+
+    # Scoped config overrides picking from the endpoint metadata.
+    if scoped_config is not None:
+        # A given service may have service specific configuration in the
+        # config file, so we need to check there as well.
+        service_config = scoped_config.get(service_name)
+        if service_config is not None and isinstance(service_config, dict):
+            version = service_config.get('signature_version')
+            if version:
+                logger.debug(
+                    "Switching signature version for service %s "
+                    "to version %s based on config file override.",
+                    service_name, version)
+                return version
+    return None
+
+
+def default_s3_presign_to_sigv2(signature_version, signing_name,
```

Comment:
That linked example isn't the same thing. It calls a function that picks a handler (or not) out of utils. Those handlers are only used in the one location as well. So it would be different. It also makes it harder to test since now we will only be able to have functional tests.